lib/client/src/swap/sanctum.rs

```diff
@@ -0,0 +1,334 @@
+        let num_loans: u8 = loan_amounts.len().try_into().unwrap();
+
+        // This relies on the fact that health account banks will be identical to the first_bank above!
+        let (health_ams, _health_cu) = self
+            .mango_client
+            .derive_health_check_remaining_account_metas(
+                account,
+                vec![source_token.token_index, target_token.token_index],
+                vec![source_token.token_index, target_token.token_index],
+                vec![],
+            )
+            .await
+            .context("building health accounts")?;
+
+        let config = self.mango_client.client.config();
+
+        let in_amount = quote
+            .in_amount
+            .clone()
+            .expect("sanctum require a in amount");
+        let in_amount_u64 = in_amount.parse::<u64>()?;
+        let out_amount =
+            ((in_amount_u64 as f64) * (1.0 - (max_slippage_bps as f64) / 10_000.0)).ceil() as u64;
+
+        let swap_response = self
+            .mango_client
+            .http_client
+            .post(format!("{}/swap", config.sanctum_url))
+            .json(&SwapRequest {
+                amount: in_amount.clone(),
+                quoted_amount: out_amount.to_string(),
+                input: input_mint.to_string(),
+                mode: "ExactIn".to_string(),
+                output_lst_mint: output_mint.to_string(),
+                signer: owner.to_string(),
+                swap_src: quote.swap_src.clone(),
+            })
+            .timeout(self.timeout_duration)
+            .send()
+            .await
+            .context("swap transaction request to sanctum")?;
+
+        let swap_r: SanctumSwapResponse = util::http_error_handling(swap_response)
+            .await
+            .context("error requesting sanctum swap")?;
+
+        let tx = bincode::options()
+            .with_fixint_encoding()
+            .reject_trailing_bytes()
+            .deserialize::<solana_sdk::transaction::VersionedTransaction>(
+                &base64::decode(&swap_r.tx).context("base64 decoding sanctum transaction")?,
+            )
+            .context("parsing sanctum transaction")?;
+
+        let (sanctum_ixs_orig, sanctum_alts) = self
+            .mango_client
+            .deserialize_instructions_and_alts(&tx.message)
+            .await?;
+
+        let system_program = system_program::ID;
+        let ata_program = anchor_spl::associated_token::ID;
+        let token_program = anchor_spl::token::ID;
+        let compute_budget_program: Pubkey = solana_sdk::compute_budget::ID;
+        // these setup instructions should be placed outside of flashloan begin-end
+        let is_setup_ix = |k: Pubkey| -> bool {
+            k == ata_program || k == token_program || k == compute_budget_program
+        };
+        let sync_native_pack =
+            anchor_spl::token::spl_token::instruction::TokenInstruction::SyncNative.pack();
+
+        // Remove auto wrapping of SOL->wSOL
```

Comment:
did you test a sanctum swap wSOL -> SOL to see if this also takes care of dropping unwrapping instructinos?

Reply:
Seems OK for both a swap to and from SOL (seen together)